steps/graph.py

```diff
@@ -1,27 +1,26 @@
 import json
-from typing import Dict, Optional
-
-from zquantum.core.graph import generate_barbell_graph as _generate_barbell_graph
-from zquantum.core.graph import generate_caveman_graph as _generate_caveman_graph
-from zquantum.core.graph import generate_graph_from_specs as _generate_graph_from_specs
-from zquantum.core.graph import generate_ladder_graph as _generate_ladder_graph
-from zquantum.core.graph import (
-    generate_random_graph_erdos_renyi as _generate_random_graph_erdos_renyi,
-)
-from zquantum.core.graph import (
-    generate_random_regular_graph as _generate_random_regular_graph,
-)
+from typing import Optional
+
+import zquantum.core.graph
 from zquantum.core.graph import save_graph
 
 
+def _make_sampler(random_weights: bool = False):
+    if random_weights:
+        return zquantum.core.graph.uniform_sampler()
+    else:
+        return zquantum.core.graph.static_sampler()
+
+
 def generate_random_graph_erdos_renyi(
     number_of_nodes: int,
     edge_probability: float,
     random_weights: bool = False,
     seed: Optional[int] = None,
 ):
-    graph = _generate_random_graph_erdos_renyi(
-        number_of_nodes, edge_probability, random_weights, seed
+    sampler = _make_sampler(random_weights)
```

Comment:
I think what would look better would be:

```
sampler = create_object(sampler_specs)()
```

Reply:
Updated in cde231f...f9d9601